presto-orc/src/main/java/com/facebook/presto/orc/reader/LongDictionaryBatchStreamReader.java

```diff
@@ -184,16 +183,11 @@ private void openRowGroup()
     {
         // read the dictionary
         if (!dictionaryOpen && dictionarySize > 0) {
-            if (dictionary.length < dictionarySize) {
-                dictionary = new long[dictionarySize];
+            int previousLength = dictionary.length;
```

Comment:
This logic tries to ensure we have enough room in the dictionary buffer and that the memory expansion is accounted. I removed the allocation of the buffer because the buffer is now allocated in LongDictionaryProvider. Actually, in the shared dictionary case the accounting would be more subtle. This is the least amount of change to still maintain correctness in this PR.